src/deepsparse/benchmark/benchmark_model.py

```diff
@@ -392,13 +394,42 @@ def benchmark_model(
         "benchmark_result": benchmark_result,
     }
 
+    parsed_export_dict = _parse_export_dict_engine_key(export_dict)
+
     # Export results
     if export_path:
         _LOGGER.info("Saving benchmark results to JSON file at {}".format(export_path))
         with open(export_path, "w") as out:
-            json.dump(export_dict, out, indent=2)
-
-    return export_dict
+            json.dump(parsed_export_dict, out, indent=2)
+
+    return parsed_export_dict
+
+
+def _parse_export_dict_engine_key(payload):
+    """Extract metadata from str(Model)"""
+    engine_split = payload["engine"].replace("\n\t", ":").split(":")
+    engine = engine_split[0]
+    payload["engine"] = engine
+    for i in range(2, len(engine_split) - 1, 2):
+        payload[engine_split[i]] = engine_split[i + 1].strip()
+    return payload
+
+
+def get_latest_commit_sha_for_path(
```

Comment:
So we're not going to have git available especially on the pip install case. The sha hash from the engine side should be printed out in the engine statement and we should just need to grab that

Reply:
The sha from the engine side is not Wand?

Reply:
With the engine startup message it prints out a sha from the wand side. This is what we want to store and report. We do not have git available when we pip install so will not be able to run any git commands.

Reply:
FWIW, it would be nice to have the length of the commit id be consistent w/ the information previously provided, so the commit id is the first 8 characters of the full commit SHA string.

Reply:
Ok tried to capture stdout in terminal in two ways. The way Im capturing does not work (perhaps its bc its a different process? Not sure how deepsparse works in the backend)

Way 1. Context Manager -- will close property even if error is induced
<img width="393" alt="Screen Shot 2022-08-08 at 10 01 12 AM" src="https://user-images.githubusercontent.com/28371594/183436631-54e3eab6-17bc-4d91-9689-a4516dc1eec3.png">

Output
<img width="992" alt="Screen Shot 2022-08-08 at 10 05 02 AM" src="https://user-images.githubusercontent.com/28371594/183436836-3014cc62-8342-4217-a4d9-07117a511728.png">

Way 2. Force output to sys stdout
<img width="386" alt="Screen Shot 2022-08-08 at 10 01 30 AM" src="https://user-images.githubusercontent.com/28371594/183436913-d2920c8c-9d39-4b29-9043-97ad50a237d8.png">

Output
<img width="1026" alt="Screen Shot 2022-08-08 at 10 01 37 AM" src="https://user-images.githubusercontent.com/28371594/183436946-0511e9f9-8e71-4c9d-9ef2-3e2f7d6493af.png">
In the output 11 doesnt  get printed (gets captured), so everything in that scope should be captured, but deepsparse bypasses this. 

TLDR: Can we just add the sha info in str(model)?

Reply:
Am I correct that this is already available to you?
```
from deepsparse import generated_version
commit_id = generated_version.revision
```

Reply:
This is what I was going to ask, all of this information parsed out from the Engine str is information we can query from the Engine class itself. Especially if you aren't finding all the information in the str representation useful, we can just extract the values needed from the Engine

Reply:
Youre a life saver thanks